### DISCUSSION TITLE
What is the difference between these two options?  `syncPolicy.automated.prune`  vs `syncPolicy.syncOptions.Prune`

### DISCUSSION THREAD

**Discussion by txjjjjj** [argoproj/argo-cd#17015] [General]
### Discussed in https://github.com/argoproj/argo-cd/discussions/16998

<div type='discussions-op-text'>

<sup>Originally posted by **txjjjjj** January 26, 2024</sup>
What is the difference between these two options?  `syncPolicy.automated.prune`  vs `syncPolicy.syncOptions.Prune`
What is the priority?

syncPolicy.automated.prune  vs syncPolicy.syncOptions.Prune
 
in Application: 
```
  syncPolicy:
    automated:
      prune: true
    syncOptions:
    - Prune=false
    - ApplyOutOfSyncOnly=true
    - RespectIgnoreDifferences=true
```</div>

**Comment by nmohoric** [comment]
For safety, automated sync doesn't prune even when you've set it to true in syncOptions (ie, syncOptions.Prune only applies to manual sync). The automated.prune setting allows you to override this and say you're aware of the downsides but want to do it anyway.

**Reply by pasha-codefresh** [reply]
Exactly! And just small note from my side , syncOptions - Prune=false not exists, you can find list of possible options here https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/
Here only PruneLast and PrunePropagationPolicy

**Reply by nerdfacerory** [reply]
Hello! I am also interested in this topic and while searching for answers i came over this conversation. @pasha-codefresh I wanted to ask what you meant with  `syncOptions - Prune=false not exists`. Do mean that Prune=False is not available as an attribute under spec.syncPolicy.syncOptions? And is only available as an annotation? Also, is there a way a user can get the list for these? `Some Sync Options can be defined as annotations in a specific resource. Most of the Sync Options are configured in the Application resource spec.syncPolicy.syncOptions attribute` It looks to me like a guesswork here. And also, the CRDs don't have these options documented. Another info that should be part of the CRDs is this question raised here: the diff between the settings under syncPolicy and the attributes under syncOptions and how these 2 interact (if they do) with each other.  From @nmohoric answer it seems they do, but i couldn't find this info in the ArgoCD docs.
